src/ComposableCow.ts

```diff
@@ -0,0 +1,113 @@
+import { ponder } from "@/generated";
+import { erc20Abi } from "../abis/erc20";
+import { decodeAbiParameters } from "viem";
+
+ponder.on("composable:ConditionalOrderCreated", async ({ event, context }) => {
+  function callERC20Contract<T>(
+    address: `0x${string}`,
+    functionName: "symbol" | "decimals" | "name"
+  ): Promise<T> {
+    return context.client.readContract({
+      abi: erc20Abi,
+      address,
+      functionName,
+    }) as Promise<T>;
+  }
+
+  function getErc20Data(address: `0x${string}`) {
+    return Promise.all([
+      callERC20Contract<string>(address, "symbol").catch(() => ""),
+      callERC20Contract<number>(address, "decimals").catch(() => 0),
+      callERC20Contract<string>(address, "name").catch(() => ""),
+    ]);
+  }
+
+  async function getToken(address: `0x${string}`) {
+    const tokenId = `${address}-${context.network.chainId}`;
+    let token = await context.db.Token.findUnique({
+      id: tokenId,
+    });
+    if (!token) {
+      const [symbol, decimals, name] = await getErc20Data(address);
+      token = await context.db.Token.create({
+        id: `${address}-${context.network.chainId}`,
+        data: {
+          address,
+          chainId: context.network.chainId,
+          symbol,
+          decimals,
+          name,
+        },
+      });
+    }
+    return token;
+  }
+
+  const userId = `${event.args.owner}-${context.network.chainId}`;
+  let user = await context.db["User"].findUnique({
+    id: userId,
+  });
+
+  if (!user) {
+    user = await context.db.User.create({
+      id: userId,
+      data: {
+        address: event.args.owner,
+        chainId: context.network.chainId,
+      },
+    });
+  }
+
+  //decode stop loss static input
```

Comment:
it would be nice to have the stop loss contract here

Reply:
what do you mean?